@xen-orchestra/rest-api/src/users/user.controller.mts

```diff
@@ -0,0 +1,59 @@
+import { Example, Get, Path, Query, Request, Response, Route, Security, Tags } from 'tsoa'
+import type { Request as ExRequest } from 'express'
+import { provide } from 'inversify-binding-decorators'
+import { notFoundResp, unauthorizedResp, type Unbrand } from '../open-api/common/response.common.mjs'
+import { XoController } from '../abstract-classes/xo-controller.mjs'
+import type { WithHref } from '../helpers/helper.type.mjs'
+import type { XoUser } from '@vates/types'
+import { partialUsers, user, userIds } from '../open-api/oa-examples/user.oa-example.mjs'
+
+@Route('users')
+@Security('*')
+@Response(unauthorizedResp.status, unauthorizedResp.description)
+@Tags('users')
+@provide(UserController)
+export class UserController extends XoController<XoUser> {
+  // --- abstract methods
+  getAllCollectionObjects(): Promise<XoUser[]> {
+    return this.restApi.xoApp.getAllUsers()
+  }
+  getCollectionObject(id: XoUser['id']): Promise<XoUser> {
+    return this.restApi.xoApp.getUser(id)
+  }
+
+  partialUser(user: XoUser): Partial<Unbrand<XoUser>> {
+    return {
+      ...user,
+      pw_hash: '***obfuscated***',
+    }
+  }
```

Comment:
The type is not valid.
`Partial<T>` means that all properties of `T` become optional.
We don't want all other pops to be optional simply because of `pw_hash`

Reply:
Only use `Unbrand<T>` when documenting an endpoint.
`Unbrand` is a type that converts `Branded<string>` to `string` to make it compatible with OpenAPI.